aten/src/ATen/test/rng_test.cpp

```diff
@@ -0,0 +1,67 @@
+#include <gtest/gtest.h>
+#include <torch/extension.h>
+#include <ATen/Generator.h>
+#include <ATen/Tensor.h>
+#include <ATen/native/cpu/DistributionTemplates.h>
+#include <ATen/core/op_registration/op_registration.h>
+
+using namespace at;
+
+namespace {
+
+constexpr uint32_t FLOAT_MASK = (1 << 24) - 1;
+constexpr float FLOAT_DIVISOR = 1.0f / (1 << 24);
+
+constexpr uint64_t DOUBLE_MASK = (1ULL << 53) - 1;
+constexpr double DOUBLE_DIVISOR = 1.0 / (1ULL << 53);
+
+struct CustomCPUGenerator : public Generator {
+  CustomCPUGenerator() : Generator{Device(DeviceType::CPU), DispatchKeySet(DispatchKey::CustomRNGKeyId)} { }
+  ~CustomCPUGenerator() = default;
+  uint32_t random() { return 42; }
+  uint64_t random64() { return 42; }
+  void set_current_seed(uint64_t seed) override { throw "not implemented"; }
+  uint64_t current_seed() const override { throw "not implemented"; }
+  uint64_t seed() override { throw "not implemented"; }
+  CustomCPUGenerator* clone_impl() const override { throw "not implemented"; }
+};
+
+Tensor& custom_rng_cauchy_(Tensor& self, double median, double sigma, Generator * generator) {
+  auto custom_generator = (CustomCPUGenerator*)generator;
+  auto iter = TensorIterator::unary_op(self, self);
+  native::templates::cauchy_kernel(iter, median, sigma, custom_generator);
+  return self;
+}
+
+template <typename T, typename RNG>
```

Comment:
Err, are these copy pastes from the `template::`? Why not just call `native::templates::cauchy_kernel` directly?